core/transport/transport.go

```diff
@@ -85,6 +85,16 @@ type Resolver interface {
 	Resolve(ctx context.Context, maddr ma.Multiaddr) ([]ma.Multiaddr, error)
 }
 
+// SkipResolver can be optionally implemented by transports that don't want to
+// resolve or transform the multiaddr. Useful for transports that wrap other
+// transports. This lets the inner transport specify how a multiaddr is
+// resolved later.
+// Also useful in cases where the transport doesn't need a resolved address to
+// dial.
+type SkipResolver interface {
+	SkipResolve(ctx context.Context, maddr ma.Multiaddr) bool
+}
```

Comment:
Can you add some text to the transport interface documentation that says something like: 

```
In addition to the Transport interface, transports may implement SkipResolver
interface. When wrapping/embedding a transport, you should ensure that 
the SkipResolver interface is handled correctly. 
```

It's very rare to wrap a transport. In case someone does, this documentation would be useful.

Reply:
I am also happy to just make `SkipResolve` a part of the transport interface.

Reply:
> I am also happy to just make SkipResolve a part of the transport interface.

I don't want to change the existing transport interface. That would be a breaking change for something that almost no transports care about.

I'll add the text to the Transport documentation however.